src/Compilers/VisualBasic/Portable/Parser/BlockContexts/SingleLineElseContext.vb

```diff
@@ -92,34 +93,90 @@ EndBlock:
                     End If
                     GoTo EndBlock
 
+                Case Else
+                    Throw ExceptionUtilities.UnexpectedValue(token.Kind)
```

Comment:
Yes, before the change we were allowing all kinds of things to pass through the Case Else.